store/tikv/2pc.go

```diff
@@ -750,33 +750,49 @@ func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *
 			// if the lock left behind whose related txn is already committed or rollbacked,
 			// (eg secondary locks not committed or rollbacked yet)
 			// we cant return "nowait conflict" directly
-			if lock.LockType == pb.Op_PessimisticLock {
```

Comment:
Why remove this check?

Reply:
I don't think the check is necessary. Blocking by an optimistic lock is possible with large transaction. (Or am I forgetting something...?)